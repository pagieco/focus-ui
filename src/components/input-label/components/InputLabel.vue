<script>

export default {
  name: 'focus-input-label',

  props: {
    /**
     * Label for the input.
     */
    label: {
      type: String,
      required: true,
    },

    /**
     * Specifies which form element a label is bound to.
     */
    labelFor: {
      type: String,
    },

    /**
     * Hides the label.
     */
    labelHidden: {
      type: Boolean,
      default: false,
    },
  },
};

</script>

<template>
  <div v-if="! labelHidden" class="input-label" :for="labelFor">
    {{ label }}
  </div>
</template>
