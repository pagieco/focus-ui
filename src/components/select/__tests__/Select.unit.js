import { mount, shallowMount } from '@vue/test-utils';
import { Select } from '../index';
import InputLabel from '../../input-label/components/InputLabel.vue';

const shallowWrap = (propsData = {}) => shallowMount(Select, { propsData });
const wrap = (propsData = {}) => mount(Select, { propsData });

describe('Select', () => {
  it('is a valid vue component', () => {
    expect(Select).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowWrap().element).toMatchSnapshot();
  });

  it('uses the disabled prop', () => {
    const wrapper = shallowWrap({ disabled: true });

    expect(wrapper.find('select').attributes('disabled')).toBe('disabled');
    expect(wrapper.find('select').attributes('aria-disabled')).toBe('true');
  });

  it('uses the id prop', () => {
    const id = 'input-id';
    const wrapper = shallowWrap({ id });

    expect(wrapper.find('select').attributes('id')).toBe(id);
  });

  it('uses the label prop', () => {
    const wrapper = shallowWrap({ label: 'Select input' });

    expect(wrapper.contains(InputLabel)).toBeTruthy();
  });

  it('uses the label-for prop', () => {
    const wrapper = wrap({ label: 'Select input', labelFor: 'my-element' });

    expect(wrapper.find(InputLabel).attributes('for')).toBe('my-element');
  });

  it('uses the label-hidden prop', () => {
    const wrapper = wrap({ label: 'Select input', labelHidden: true });

    expect(wrapper.find(InputLabel).html()).toBeUndefined();
  });

  it('uses the name prop', () => {
    const name = 'input-name';
    const wrapper = shallowWrap({ name });

    expect(wrapper.find('select').attributes('name')).toBe(name);
  });

  it('uses the options prop', () => {
    const wrapper = wrap({
      options: [
        { value: 123, label: 'My label' },
        { value: 456, label: 'Another label' },
      ],
    });

    expect(wrapper.findAll('option').length).toBe(2);
  });

  it('uses the placeholder prop', () => {
    const wrapper = wrap({ placeholder: 'Placeholder text' });

    expect(wrapper.find('option[disabled][selected][value=""]').exists()).toBeTruthy();
  });
});
