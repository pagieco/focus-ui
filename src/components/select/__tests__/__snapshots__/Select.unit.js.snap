// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select matches the snapshot 1`] = `
<div
  class="select-input"
>
  <!---->
   
  <select>
    <!---->
     
  </select>
   
  <div
    class="select-input__content"
  >
    <div
      class="select-input__selected-option"
    >
      
    </div>
     
    <div
      class="select-input__icon"
    />
  </div>
</div>
`;
